storage-bigtable/src/lib.rs

```diff
@@ -550,14 +551,14 @@ impl LedgerStorage {
     }
 
     // Fetches and gets a vector of confirmed blocks via a multirow fetch
-    pub async fn get_confirmed_blocks_with_data<'a>(
+    pub async fn get_confirmed_blocks_with_data(
         &self,
-        slots: &'a [Slot],
-    ) -> Result<impl Iterator<Item = (Slot, ConfirmedBlock)> + 'a + use<'a>> {
+        slots: impl IntoIterator<Item = Slot> + Debug,
```

Comment:
Hmm, I think we should probably remove `{slots:?}` from the output + drop the ` + Debug`; I don't think it is going to give useful output if it is an iterator will it ? Maybe a quick playground test could confirm

Reply:
now the function receives `HashSet` (as an `IntoIterator + Debug`), so it will in fact provide clean debug string, but surely just dropping that from `trace` might be fine..

Reply:
Hmm ok, I figured the `Debug` would give us some hacked up for the type (Iterator, Map, etc) but a quick playground link confirms it gives us nice output:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=18ff3612e5963807dfab18891769b3cd

So that will iterate through, but only if the log level (trace in this case) is actually enabled here so I think it is fine to leave it; this sound right to you ?